vault/resource_database_secret_backend_connection.go

```diff
@@ -1033,6 +1051,29 @@ func getMSSQLConnectionDetailsFromResponse(d *schema.ResourceData, prefix string
 	return result, nil
 }
 
+func getPostgresConnectionDetailsFromResponse(d *schema.ResourceData, prefix string, resp *api.Secret) map[string]interface{} {
+	result := getConnectionDetailsFromResponseWithDisableEscaping(d, prefix, resp)
+	details := resp.Data["connection_details"]
+	data, ok := details.(map[string]interface{})
+	if !ok {
+		return nil
+	}
+
+	// cloud specific
```

Comment:
I would have liked to not duplicate this code - similarly to how the set versions call a cloud-specific function, but because of the way the results chain, the function here wouldn't match the other getter function pattern.

It still might be worth it, I suppose - we'd either add a new parameter to branch what getConnectionDetails function we call inside, or pass the result map 'by reference' and set the values outside the usual get-details chain.

It has to be this way, because postgres and mysql call different setters (i.e., postgres calls 'WithDisableEscaping', while MySql calls WithUserPass directly.